contribute/design/cstor-operator/specs.md

```diff
@@ -0,0 +1,438 @@
+  // Possible values are : auto or manual
+  // This field is required.
+  provisioningType: auto
+
+  // maxPools is the desired number of pool that should be provisioned.
+  // This field is required.
+  maxPools: 3
+
+  // type is the raid group Possible values : stripe, mirror, raidz or raidz2.
+  // This field is required.
+  type: raidz 
+
+  // cache file is used put pool configuration which can make the import faster.
+  // This field is optional.
+  cacheFile: var/openebs/img-0
+  
+  // overProvisioning if set to true allows over provisioning else does not.
+  // This field is optional i.e. by default it will be false.
+  overProvisioning: false
+
+  // disk type is the type of disk e.g. sparse or disk
+  // This field is required. This is honoured only in auto provisioning.
+  diskType: sparse 
+
+
+```
+
+Notes: The extra blank lines in above YAML spec is provided just for readability.
+
+
+Notes:
+1. Striped PoolType cannot have more than 1 group.
+
+2. CStor-Operator will not reconcile for older StoragePoolClaim ( SPC ) once the
+   code to reconcile for CStorPoolCluster is checked in.
+
+Above YAML specs are just few sample YAML. To understand a complete exhaustive
+spec for CStorPoolCluster, following is the proposed API spec in go struct:
+
+```
+// CStorPoolCluster describes a CStorPoolCluster custom resource.
+type CStorPoolCluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              CStorPoolClusterSpec   `json:"spec"`
+	Status            CStorPoolClusterStatus `json:"status"`
+}
+
+// CStorPoolClusterSpec is the spec for a CStorPoolClusterSpec resource
+type CStorPoolClusterSpec struct {
+
+
+    // MaxPools is the desired pool count in case of auto provisioning.
+    MaxPools *int        
+               `json:"maxPools"`
+    // diskType is the disk type -- e.g. sparse or disk
+    // This field is required but is not honoured in manual provisioning.
+    diskType     string                     `json:"diskType"`
+
+    // cache file is used put pool configuration which can make the import faster.
+    // This field is optional.
+    CacheFile        string   `json:"cacheFile"`    
+ 
+    // type is the raid group Possible values : stripe, mirror, raidz or raidz2.
+    
+    // stripe: It is a raid group which divides data into blocks and spreads the 
+    // data blocks across multiple block devices.
+
+    // mirror: It is a raid group which creates redundancy of data across block devices. 
+     
+    // This field is required.
+    
+    Type         PoolType `json:"poolType"`    
+ 
+    // overProvisioning if set to true allows over provisioning else does not.
+    // This field is optional i.e. by default it will be false.
+    OverProvisioning bool     `json:"overProvisioning"` //true or false
+
+    // Nodes is the list of node spec which contains details for pool
+    // provisioning on the nodes.
+    Nodes    []CStorPoolClusterNodeSpec `json:"nodes"`
```

Comment:
This can be really called a cStorPoolSpecs -- because it defines the cStorPools on a given node. (Given this flow, we might actually have to change the main sub-section from "nodes" -> "pools" -- in which case the nodeSelector concept will fit in.